### DISCUSSION TITLE
Designing Table, Modal Popup Box, Menu Responsiveness

### DISCUSSION THREAD

**Discussion by asmsaiff** [tailwindlabs/tailwindcss#5684] [Ideas]
Hey, This is Saif, a full-stack web developer from Dhaka, Bangladesh. I'm using TailwindCSS as my 1st priority over any other css framework/library as well and It's really superb. 

But I've faced some issues when I design tables, popup modal box and responsive menu. Sometimes I use Alpine.js and most of the time I write my won script for designing a modal or clickable side panel/toggle responsive menu.  It's very time depending. So I switched bootstrap sometimes for this kind of job. 

Is there any built in way to design table, responsive menu and popup modal with TailwindCSS ????
Regards.

**Comment by Nikit-Singh** [comment]
You can try using [HeadlessUI](https://headlessui.dev/) and [React-Table](https://github.com/tannerlinsley/react-table) for these.
